### ISSUE TITLE
DeprecatedError usability concerns

### ISSUE THREAD

**Issue by snappizz** [supercollider/supercollider#2868]
Class methods have ugly DeprecatedError output:

```
OSCresponder.deprecated(OSCresponder.class.findMethod(\new), OSCFunc.class.findMethod(\new))'
-> WARNING: Called from {unknown}, method Meta_OSCresponder:new is deprecated and will be removed. Use Meta_OSCFunc:new instead.
```

Should be `OSCresponder.new` and `OSCFunc.new`.

The second issue is that the structure of DeprecatedError is too narrow. Correct me if I'm wrong, but I don't think you can recommend a method inherited from a superclass. We can't recommend multiple methods, and we can't recommend something that isn't a method (e.g. a quark or a helpfile). It would be best if we had the option to supply our own message to the user.

**Comment by telephon** [comment]
Yes, I always found this limiting. Somehow, a string would be better - some methods have no direct replacement.

**Comment by snappizz** [comment]
To be pedantic, the class also seems misnamed. DeprecatedError only behaves as an actual error when `Error.debug` is true, and that is the exception, not the norm. Too late to change it, I guess.

**Comment by brianlheim** [comment]
> Should be OSCresponder.new and OSCFunc.new.

Not sure I understand.. `new` is a static method, not an instance method.

**Comment by brianlheim** [comment]
Would a viable solution be to allow the `alternateMethod` argument to be a string? Is the format `"Use x instead."` also too limiting?

**Comment by snappizz** [comment]
> Not sure I understand.. `new` is a static method, not an instance method.

Yes, I said it's a class method... what about it?

> Would a viable solution be to allow the alternateMethod argument to be a string? Is the format "Use x instead." also too limiting?

Sounds good. If it's a Method instance, use the current behavior, and if it's a string, replace the "use..." sentence.

**Comment by muellmusik** [comment]
Meta_SomeClass:someMethod is the way that SC labels class methods. Could be more intuitively user readable, but still good to indicate it's a class method, as some classes (e.g. TempoClock) have class and instance methods with the same names.

If you'd like to replace or rename this, I don't see any issue really. Just deprecate DeprecatedError. ;-)

More seriously, since the Objectdeprecated method is the normal way of accessing this (I don't think anyone should be creating DeprecatedErrors directly) you can just adjust the implementation if you like providing the interface stays the same. You could have a separate deprecateBlock a la Smalltalk, or check for debug, or an alternate method which allows a string. The latter might be cleanest. (It's possible Quarks and user lib code use deprecated.)

**Comment by telephon** [comment]
or you can just add a string argument, and ignore the others if not given
